sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/SyntaxTrackers/ObjectCreationTrackerTest.cs

```diff
@@ -0,0 +1,99 @@
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+using SonarAnalyzer.UnitTest.TestFramework;
+using CSharpSyntax = Microsoft.CodeAnalysis.CSharp.Syntax;
+using VBSyntax = Microsoft.CodeAnalysis.VisualBasic.Syntax;
+
+namespace SonarAnalyzer.UnitTest.Helpers
+{
+    [TestClass]
+    public class ObjectCreationTrackerTest
+    {
+        [TestMethod]
+        public void ConstArgumentForParameter_CS()
+        {
+            var testInput = @"
+namespace NS
+{
+  public class Base
+  {
+    private Base(string a, string b, bool c, int d, object e) {}
+
+    private void Usage(string notAConst)
+    {
+      new Base(notAConst, ""myConst"", true, 4, new object());
+    }
+  }
+}
+";
+            var context = CreateContext<CSharpSyntax.ObjectCreationExpressionSyntax>(testInput, AnalyzerLanguage.CSharp);
+            var tracker = new CSharpObjectCreationTracker(null, null);
+
+            tracker.ConstArgumentForParameter(context, "a").Should().BeNull();
+            tracker.ConstArgumentForParameter(context, "b").Should().Be("myConst");
+            tracker.ConstArgumentForParameter(context, "c").Should().Be(true);
+            tracker.ConstArgumentForParameter(context, "d").Should().Be(4);
+            tracker.ConstArgumentForParameter(context, "e").Should().BeNull();
+            tracker.ConstArgumentForParameter(context, "nonExistingParameterName").Should().BeNull();
+        }
+
+        [TestMethod]
+        public void ConstArgumentForParameter_VB()
+        {
+            var testInput = @"
+Namespace NS
+    Public Class Base
+        Sub New(ByVal a As String, b As String, ByVal c As Boolean, ByVal d As Integer, ByRef e As Integer, ByVal f As Object)
+        End Sub
+
+        Public Sub Usage(ByVal notAConst As String)
+            Dim tmp = New Base(notAConst, ""myConst"", True, 4, 5, New Object())
+        End Sub
+    End Class
+End Namespace
+";
+            var context = CreateContext<VBSyntax.ObjectCreationExpressionSyntax>(testInput, AnalyzerLanguage.VisualBasic);
+            var tracker = new VisualBasicObjectCreationTracker(null, null);
+
+            tracker.ConstArgumentForParameter(context, "a").Should().BeNull();
+            tracker.ConstArgumentForParameter(context, "b").Should().Be("myConst");
+            tracker.ConstArgumentForParameter(context, "c").Should().Be(true);
+            tracker.ConstArgumentForParameter(context, "d").Should().Be(4);
+            tracker.ConstArgumentForParameter(context, "e").Should().Be(5);
+            tracker.ConstArgumentForParameter(context, "f").Should().BeNull();
+            tracker.ConstArgumentForParameter(context, "nonExistingParameterName").Should().BeNull();
+        }
+
+        private static ObjectCreationContext CreateContext<TSyntaxNodeType>(string testInput, AnalyzerLanguage language) where TSyntaxNodeType : SyntaxNode
+        {
+            var testCode = new SnippetCompiler(testInput, ignoreErrors: false, language: language);
```

Comment:
Do we need to be explicit about parameter names? They are in order and overloads are simple here.
```suggestion
            var testCode = new SnippetCompiler(testInput, false, language);
```